pkg/steps/git_source.go

```diff
@@ -0,0 +1,140 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	coreapi "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	buildapi "github.com/openshift/api/build/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+
+	"github.com/openshift/ci-operator/pkg/api"
+)
+
+type gitSourceStep struct {
+	gitConfig   api.InputGitSourceStepConfiguration
+	resources   api.ResourceConfiguration
+	imageClient imageclientset.ImageV1Interface
+	buildClient BuildClient
+	jobSpec     *JobSpec
+}
+
+func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return s.jobSpec.Inputs(), nil
+}
+
+func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
+	return handleBuild(s.buildClient, buildFromGitSource(s.jobSpec, s.gitConfig, s.gitConfig.To, s.resources), dry)
+}
+
+func (s *gitSourceStep) Done() (bool, error) {
+	return imageStreamTagExists(s.gitConfig.To, s.imageClient.ImageStreamTags(s.jobSpec.Namespace()))
+}
+
+func (s *gitSourceStep) Name() string { return string(s.gitConfig.To) }
+
+func (s *gitSourceStep) Description() string {
+	return fmt.Sprintf("Build git source code into an image and tag it as %s", s.gitConfig.To)
+}
+
+func (s *gitSourceStep) Requires() []api.StepLink { return nil }
+
+func (s *gitSourceStep) Creates() []api.StepLink {
+	return []api.StepLink{api.InternalImageLink(s.gitConfig.To)}
+}
+
+func (s *gitSourceStep) Provides() (api.ParameterMap, api.StepLink) {
+	return api.ParameterMap{
+		"BUILD_ROOT_IMAGE": func() (string, error) {
```

Comment:
Still not correct, this should be the name of the parameter expected by a template step.